### PR TITLE
Fix(BBD-838): Mouse into Sayt, click out, try to search, doesn't work

### PR DESCRIPTION
**Pull request by treeforever**
Please review along with this PR in preset: https://github.com/groupby/storefront-sayt/pull/38

### CODE REVIEW

File: test/unit/services/autocomplete.ts [hunk 1]
```diff
@@ -70,11 +70,21 @@ suite('Autocomplete Service', ({ expect, spy, stub, itShouldBeCore, itShouldExte
...
+      const on = (e, cb) => {
+        if (e === 'unmount') {
+          unmountCb = cb;
+        }
+      };
+      const tag: any = { a: 'b', on };
 
       service.registerAutocomplete(tag);
-
+
```
[the remark this answers is not in the document]
Comment (zfcb) [reply test/unit/services/autocomplete.ts]: Would a spy() + args pattern work better here?

File: test/unit/services/autocomplete.ts [hunk 2]
```diff
@@ -70,11 +70,21 @@ suite('Autocomplete Service', ({ expect, spy, stub, itShouldBeCore, itShouldExte
...
+      const tag: any = { a: 'b', on };
 
       service.registerAutocomplete(tag);
-
+      
       expect(service.registeredAutocompleteTags).to.eql([tag]);
+      
+      unmountCb();
+      
+      expect(service.registeredAutocompleteTags).to.eql([]);
```
Comment (Filip-F) [inline test/unit/services/autocomplete.ts]: I don't think you need this much spacing between statements in the test.

File: test/unit/services/autocomplete.ts [hunk 3]
```diff
@@ -70,11 +70,21 @@ suite('Autocomplete Service', ({ expect, spy, stub, itShouldBeCore, itShouldExte
 
   describe('registerAutocomplete()', () => {
     it('should add tag to registeredAutocompleteTags', () => {
-      const tag: any = { a: 'b' };
+      let unmountCb;
+      const on = (e, cb) => {
```
Comment (Filip-F) [inline test/unit/services/autocomplete.ts]: You don't need to do all of this, you can do 
```
on = spy();
...
on.getCall(0).args[1]
```
in order to get the function passed into on.

File: src/services/autocomplete.ts [hunk 4]
```diff
@@ -29,6 +29,9 @@ class AutocompleteService extends LazyService {
 
   registerAutocomplete(tag: AutocompleteTag) {
     this.registeredAutocompleteTags.push(tag);
+    tag.on('unmount', () => {
```
Comment (Vic-Dev) [inline src/services/autocomplete.ts]: Can be one line

**Pull request merged**